xtreeshr/XTreeGetTimedRecord.c

```diff
@@ -312,6 +312,12 @@ EXPORT int _XTreeGetTimedRecord(void *dbid, int inNid, struct descriptor *startD
 		status = (dbid)?_TreeGetSegment(dbid, nid, currIdx, &dataXds[currSegIdx], &dimensionXds[currSegIdx]):TreeGetSegment(nid, currIdx, &dataXds[currSegIdx], &dimensionXds[currSegIdx]);
 
 //printf("Read Segment %d\n", currSegIdx);
+		if STATUS_OK
+		{
+		  if (dimensionXds[currSegIdx].pointer->dtype == DTYPE_FUNCTION) {
+		    status = TdiData(dimensionXds[currSegIdx].pointer,&dimensionXds[currSegIdx] MDS_END_ARG);
```

Comment:
i would not check for FUNCTION but for the class.
or call data in any case which could be a NOP if the argument is already a simple type (array or scalar)

Reply:
I changed it to always do TdiData() for the dim.